weather/weather_test.go

```diff
@@ -1,14 +1,55 @@
 package weather
 
 import (
+	"os"
 	"testing"
 
+	"github.com/jarcoal/httpmock"
 	"github.com/stretchr/testify/assert"
 )
 
+var (
+	apiKey   string
+	testCity string
+
+	testClient *WeatherClient
+)
+
+func TestMain(m *testing.M) {
+	apiKey = "apikey"
+	testCity = "testville"
+
+	config := WeatherConfig{
+		ApiKey: apiKey,
+		Units:  "metric",
+	}
+
+	testClient, _ = NewClient(config)
+
+	httpmock.ActivateNonDefault(testClient.httpClient.GetClient())
+	defer httpmock.DeactivateAndReset()
+
+	os.Exit(m.Run())
+}
+
 func TestNewClient(t *testing.T) {
-	client := NewClient("apikey", "metric")
+	assert.Equal(t, "metric", testClient.unit)
+}
+
+func TestNewClientWithoutApiUnits(t *testing.T) {
+
+	config := WeatherConfig{ApiKey: "apikey"}
+	testClient, err := NewClient(config)
+
+	assert.Equal(t, "metric", testClient.unit)
+	assert.Nil(t, err)
```

Comment:
```suggestion
	assert.NoError(t, err)
```
Accomplishes the same thing, but reads clearer.